plugins/tar/commands

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+set -eo pipefail; [[ $DOKKU_TRACE ]] && set -x
+source "$(dirname $0)/../common/functions"
+
+case "$1" in
+  tar-from|tar:from)
+    verify_app_name "$2"
+    APP=$2
+    URL=$3
+    shift 3
+    curl -# --insecure -L "$URL" | dokku tar-in $APP $@
+    ;;
+
+  tar-in|tar:in)
+    APP="$2"
+    verify_app_name "$2"
+    tee "$DOKKU_ROOT/$APP/src.tar" | wc -c 
+    pluginhook -p receive-app $APP
+    ;;
+
+  tar-build)
+    verify_app_name "$2"
+    APP="$2"; APP_BUILD_LOCK="$DOKKU_ROOT/$APP/.build.lock"
+    APP_BUILD_LOCK_MSG="$APP is currently being deployed or locked. Waiting..."
+    [[ $(flock -n "$APP_BUILD_LOCK" true &>/dev/null ; echo $?) -ne 0 ]] && echo "$APP_BUILD_LOCK_MSG"
+
+    shift 1
+    flock -o "$APP_BUILD_LOCK" dokku tar-build-locked "$@"
+    ;;
+
+  tar-build-locked)
+    APP="$2"
+    verify_app_name "$2"
+    shift 2
+
+    # clean up after ourselves
+    TMP_WORK_DIR=$(mktemp -d)
+    trap 'rm -rf "$TMP_WORK_DIR" > /dev/null' RETURN
+
+    # extract tar file
+    chmod 755 $TMP_WORK_DIR
+    pushd $TMP_WORK_DIR > /dev/null
+    tar -x -C "$TMP_WORK_DIR" -f "$DOKKU_ROOT/$APP/src.tar"  
+    #find -name .git -prune -exec rm -rf {} \; > /dev/null
+
+    if [[ -f Dockerfile ]] && [[ "$([[ -f .env ]] && grep -q BUILDPACK_URL .env; echo $?)" != "0" ]] && [[ ! -f ".buildpacks" ]]; then
+      dokku receive "$APP" "dockerfile" "$TMP_WORK_DIR" | sed -u "s/^/"$'\e[1G'"/"
+    else
+      dokku receive "$APP" "herokuish" "$TMP_WORK_DIR" | sed -u "s/^/"$'\e[1G'"/"
+    fi
+    ;;
```

Comment:
This bit is pretty much copied from the git plugin, the mercurial plugin copies it too.  Should this be moved into common?